src/ui/ServerAdministration/AdminRealm.ts

```diff
@@ -0,0 +1,49 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2018 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { createRealmContext } from 'react-realm-context';
+import * as Realm from 'realm';
+
+const {
+  RealmProvider: AdminRealmProvider,
+  RealmConsumer: AdminRealmConsumer,
+  RealmQuery: AdminRealmQuery,
+  RealmConnection: RealmAdminConnection,
+  withRealm: withAdminRealm,
+} = createRealmContext();
+
+export function getAdminRealmConfig(
+  user: Realm.Sync.User,
+  partialConfig: Realm.PartialConfiguration = {},
+) {
+  const config = user.createConfiguration(partialConfig);
+  // TODO: Simplify this once https://github.com/realm/realm-js/issues/1981 is fixed
+  if (config.sync && config.sync.url) {
```

Comment:
Is there a chance that this is not a sync configuration? I don't think `user.createConfiguration` can ever return a local-only config.

Reply:
This if-condition is for type inference, `user.createConfiguration` just returns a `RealmConfiguration` which provides no guarantees on the existence of `config.sync`.

Reply:
Seems a bit like bending over backwards to satisfy the type checker. It doesn't make sense for `user.createConfiguration` to give us a local config as local configs don't need a user. So I would rather trust the API and force the type checker to cooperate than to write non-obvious code - e.g. I was legitimately confused when reading this about how this method works to be able to get an empty sync config.

Reply:
If you can propose a change in the typescript definitions in realm-js to produce objects with correct nullability definitions, I'm fine with that, but ultimately, it's not the job of the code in the studio repo to verify that realm-js's API are behaving correctly.

Reply:
I agree - tightening the Realm JS types is the right thing to do.

Reply:
I've created https://github.com/realm/realm-js/issues/2163 to track this.